src/core/observable.ts

```diff
@@ -106,7 +106,11 @@ export function startBatch() {
 }
 
 export function endBatch() {
-    if (--globalState.inBatch === 0) {
+    if (
+        --globalState.inBatch === 0 ||
+        globalState.pendingReactions.length > 0 ||
```

Comment:
This breaks all batch logic! In fact, batches become useless after this change.

Reply:
yes, that's the outcome to my understanding, too. Considering performance test results and tests in real-life app that easily generates ~50 actions a second, how are the batches still useful? For us, this is currently causing more pain than any measurable difference known to me. What are some other benefits of batches I'm not seeing?

Reply:
Simple ObservableMap,set call (and any observable object property assignment!) can generate 3 different changes in one batch. You cannot maintain model consistency without batches.

Reply:
Makes sense. Happy to take this to the original issue if you're aware of any ways to perform proper debugging in order to find the core issue. Otherwise we have to run mobx from our fork without the proper model consistency.